### PR TITLE
fix($location): $locationchangesuccess not fires when browser location changes to URL which ends with# issue #12175  

### PR DESCRIPTION
**Pull request by rrsivabalan**
fix($location): $locationchangesuccess not fires when browser location changes to URL which ends with# issue #12175

### CODE REVIEW

**Comment by rrsivabalan** [comment]
@petebacondarwin  I have fixed code review provided in PR https://github.com/angular/angular.js/pull/12862  and also added UT to demonstrate  the bug #12175 
Can you please review the changes

**Comment by rrsivabalan** [comment]
@petabacondarwin can you please review changes and provide your comments.

File: src/ng/location.js [hunk 1]
```diff
@@ -918,16 +918,16 @@ function $LocationProvider() {
         var oldUrl = $location.absUrl();
         var oldState = $location.$$state;
         var defaultPrevented;
-
+		var trimNewUrl = trimEmptyHash(newUrl);
```
Comment (petebacondarwin) [inline src/ng/location.js]: spaces for tabs

**Comment by rrsivabalan** [comment]
@petebacondarwin  thanks for reviewing and accepting code changes to close the issue #12175

**Pull request closed without merging**